packages/machine-extractor/src/__tests__/parseResult.test.ts

```diff
@@ -138,57 +138,79 @@ describe('MachineParseResult', () => {
     });
   });
 
-  it('should extract inline custom action', () => {
+  it('should extract inline expressions as inline actions', () => {
     const config = getTestMachineConfig(
       `
     createMachine({
       initial: "a",
       states: {
         a: {
-          entry: [() => {
-            console.log('test')
-          }],
-          exit: [function() {}],
-        },
-        b: {
-          entry: [someVar],
+          entry: [() => {}],
+          exit: [someVar]
         }
       },
     });
   `,
     );
     expect(config.states!.a.entry).toMatchInlineSnapshot(`
-      {
-        "type": "xstate.custom",
-        "value": {
-          "type": "expression",
-          "value": "() => {
-                  console.log('test')
-                }",
+      [
+        {
+          "action": {
+            "expr": "{{() => {}}}",
+          },
+          "kind": "inline",
         },
-      }
+      ]
     `);
     expect(config.states!.a.exit).toMatchInlineSnapshot(`
-      {
-        "type": "xstate.custom",
-        "value": {
-          "type": "expression",
-          "value": "function() {}",
+      [
+        {
+          "action": {
+            "expr": "{{someVar}}",
+          },
+          "kind": "inline",
         },
-      }
+      ]
     `);
-    expect(config.states!.b.entry).toMatchInlineSnapshot(`
-      {
-        "type": "xstate.custom",
-        "value": {
-          "type": "expression",
-          "value": "someVar",
+  });
+
+  it('should extract action objects as inline actions', () => {
+    const config = getTestMachineConfig(
+      `
+    createMachine({
+      initial: "a",
+      states: {
+        a: {
+          entry: [{type: someIdentifier, params: {foo: 'bar'}}],
+          exit: [{type: 'xstate.assign', assignment: {foo: 'bar', baz: () => {}}}]
+        }
+      },
+    });
+  `,
+    );
+    expect(config.states!.a.entry).toMatchInlineSnapshot(`
+      [
+        {
+          "action": {
+            "expr": "{{{type: someIdentifier, params: {foo: 'bar'}}}}",
```

Comment:
Is this just because we can't/don't extract the actual object yet? This should ideally be a normal action object inside of `action`, not an expression

Reply:
Oh I see, it's because of `someIdentifier`, sorry!